pages/badges/badge-single.jsx

```diff
@@ -0,0 +1,449 @@
+'use strict';
+
+var React = require('react'),
+    Link = require('react-router').Link,
+    LoginLink = require('../../components/login/LoginLink.jsx'),
+
+    RequirementsList = require('../../components/badges/requirement-list.jsx'),
+    SocialShare = require('../../components/badges/social-share.jsx'),
+    Badge = require('../../components/badges/badge.jsx'),
+    BadgeHorizontalIcon = require('../../components/badges/badge-horizontal-icon.jsx'),
+    CredlyLinkForm = require('../../components/badges/CredlyLinkForm.jsx'),
+
+    Divider = require('../../components/Divider.jsx'),
+    BadgesAPI = require('../../lib/badges-api'),
+    TeachAPI = require('../../lib/teach-api'),
+    Modal = require('../../components/modal.jsx');
+
+var Navigation = React.createClass({
+  render: function() {
+    var prev = this.props.prev;
+    var next = this.props.next;
+    return (
+      <div className="badge-navigation">
+        <a className="previous" href={prev.url}>
+          <img src={prev.img} />
+          <span clasName="label">← {prev.title}</span>
+        </a>
+        <a className="next" href={next.url}>
+          <span clasName="label">{next.title} →</span>
+          <img src={next.img} />
+        </a>
+      </div>
+    );
+  }
+});
+
+/**
+ * Single badge Page
+ * should take a badgeId
+ */
+var BadgePage = React.createClass({
+  statics: {
+    pageTitle: 'Badges',
+    pageClassName: 'badges single-badge'
+  },
+
+  contextTypes: {
+    history: React.PropTypes.object
+  },
+
+  getInitialState: function () {
+    var teachAPI = this.props.teachAPI || new TeachAPI();
+    var badgeAPI = new BadgesAPI({ teachAPI: teachAPI });
+    return {
+      hasAccess: false,
+      showLinkModal: false,
+      applying: false,
+      showApplyModal: false,
+      teachAPI: teachAPI,
+      badgeAPI: badgeAPI,
+      badge: {
+        id: "",
+        title: "",
+        status: '',
+        description: "",
+        icon: "",
+        icon2x: "",
+        criteria: []
+      },
+      prev: false,
+      next: false,
+      evidenceLink: '',
+      evidenceText: '',
+      evidenceFiles: []
+    };
+  },
+
+  toggleAccess: function(err, result) {
+    result = result || { access: false };
+    this.setState({
+      hasAccess: result.access
+    });
+  },
+
+  componentDidMount: function() {
+    // get this specific badge's details
+    this.state.badgeAPI.getBadgeDetails(this.props.params.id, this.handleBadgeData);
+
+    // we're also interested in whether this user is credly-authenticated
+    this.state.badgeAPI.hasAccess(this.toggleAccess, function(err, data) {
+      if (err) return console.error("not logged into credly");
```

Comment:
Should we introduce some UI changes based on user's credly authentication state?